File: app/src/main/java/com/princeakash/projectified/HomeFragment.java
package com.princeakash.projectified;

import android.os.Bundle;

import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.util.ArrayList;

/**
 * A simple {@link Fragment} subclass.
 * Use the {@link HomeFragment#newInstance} factory method to
 * create an instance of this fragment.
 */
public class HomeFragment extends Fragment implements HomeAdapter.HomeListener {

    private ArrayList<HomeAdapter.HomeItem> list;

    // TODO: Rename parameter arguments, choose names that match
    // the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
    private static final String ARG_PARAM1 = "param1";
    private static final String ARG_PARAM2 = "param2";

    // TODO: Rename and change types of parameters
    private String mParam1;
    private String mParam2;

    public HomeFragment() {
        // Required empty public constructor
    }

    /**
     * Use this factory method to create a new instance of
     * this fragment using the provided parameters.
     *
     * @param param1 Parameter 1.
     * @param param2 Parameter 2.
     * @return A new instance of fragment HomeFragment.
     */
    // TODO: Rename and change types and number of parameters
    public static HomeFragment newInstance(String param1, String param2) {
        HomeFragment fragment = new HomeFragment();
        Bundle args = new Bundle();
        args.putString(ARG_PARAM1, param1);
        args.putString(ARG_PARAM2, param2);
        fragment.setArguments(args);
        return fragment;
    }



    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            mParam1 = getArguments().getString(ARG_PARAM1);
            mParam2 = getArguments().getString(ARG_PARAM2);
        }
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view = inflater.inflate(R.layout.frag_home, container, false);
        RecyclerView recyclerView = view.findViewById(R.id.recyclerView);

        list = new ArrayList<>();
        list.add(new HomeAdapter.HomeItem("App Development", R.mipmap.app_dev));
        list.add(new HomeAdapter.HomeItem("Web Development", R.mipmap.web));
        list.add(new HomeAdapter.HomeItem("Machine Learning", R.mipmap.ml));
        list.add(new HomeAdapter.HomeItem("Artificial Intelligence", R.mipmap.artificial_intelligence));
        list.add(new HomeAdapter.HomeItem("Content Writing", R.mipmap.content));
        list.add(new HomeAdapter.HomeItem("UI/UX Design", R.mipmap.ui));
        list.add(new HomeAdapter.HomeItem("Video Editing", R.mipmap.video));
        list.add(new HomeAdapter.HomeItem("Others", R.mipmap.misc));

        recyclerView.setAdapter(new HomeAdapter(list, getContext(), this));
        recyclerView.setLayoutManager(new GridLayoutManager(getContext(), 2, RecyclerView.VERTICAL, false));
        return view;
    }
}

File: app/src/main/java/com/princeakash/projectified/candidate/addApplication/view/HomeFragment.kt
package com.princeakash.projectified.candidate.addApplication.view

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import androidx.navigation.findNavController
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.princeakash.projectified.R
import com.princeakash.projectified.candidate.addApplication.view.HomeAdapter.Companion.AI
import com.princeakash.projectified.candidate.addApplication.view.HomeAdapter.Companion.ANDROID
import com.princeakash.projectified.candidate.addApplication.view.HomeAdapter.Companion.CONTENT
import com.princeakash.projectified.candidate.addApplication.view.HomeAdapter.Companion.ML
import com.princeakash.projectified.candidate.addApplication.view.HomeAdapter.Companion.OTHERS
import com.princeakash.projectified.candidate.addApplication.view.HomeAdapter.Companion.UIUX
import com.princeakash.projectified.candidate.addApplication.view.HomeAdapter.Companion.VIDEO
import com.princeakash.projectified.candidate.addApplication.view.HomeAdapter.Companion.WEB
import com.princeakash.projectified.candidate.addApplication.view.HomeAdapter.HomeItem
import com.princeakash.projectified.candidate.addApplication.view.HomeAdapter.HomeListener
import com.princeakash.projectified.candidate.myApplications.viewModel.CandidateViewModel
import kotlinx.android.synthetic.main.frag_home.view.*

//import java.util.*


class HomeFragment : Fragment(), HomeListener {

    private lateinit var list: ArrayList<HomeItem>
    private lateinit var candidateAddApplicationsViewModel: CandidateViewModel

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        val view = inflater.inflate(R.layout.frag_home, container, false)
        val recyclerView: RecyclerView = view.findViewById(R.id.recyclerView)
        list = ArrayList()
        list.add(HomeItem("App Development", R.mipmap.app_dev,ANDROID))
        list.add(HomeItem("Web Development", R.mipmap.web, WEB))
        list.add(HomeItem("Machine Learning", R.mipmap.ml, ML))
        list.add(HomeItem("Artificial Intelligence", R.mipmap.artificial_intelligence, AI))
        list.add(HomeItem("Content Writing", R.mipmap.content, CONTENT))
        list.add(HomeItem("UI/UX Design", R.mipmap.ui, UIUX))
        list.add(HomeItem("Video Editing", R.mipmap.video, VIDEO))
        list.add(HomeItem("Others", R.mipmap.misc, OTHERS))
        recyclerView.adapter = context?.let { HomeAdapter(list, it, this) }
        recyclerView.layoutManager = GridLayoutManager(context, 2, RecyclerView.VERTICAL, false)
        return view
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        candidateAddApplicationsViewModel = ViewModelProvider(requireActivity()).get(CandidateViewModel::class.java)

        candidateAddApplicationsViewModel.safeToVisitDomainOffers().observe(viewLifecycleOwner, {
            it.getContentIfNotHandled()?.let{safeToVisit->
                if(safeToVisit){
                    view.findNavController().navigate(R.id.home_to_offers_by_domain)
                }
            }
        })
    }

    override fun onResume() {
        super.onResume()
        (requireActivity() as AppCompatActivity).supportActionBar?.title = "Projectified"
    }

    override fun onDomainClick(position: Int) {
        requireView().progress_circular_layout.visibility = View.VISIBLE
        val domainArg = list[position].domainArg
        candidateAddApplicationsViewModel.getOffersByDomain(domainArg)
    }
}